homeassistant/components/shelly/sensor.py

```diff
@@ -72,6 +74,28 @@
 PARALLEL_UPDATES = 0
 
 
+def get_entity_translation_attributes(
+    channel_name: str | None,
+    translation_key: str | None,
+    device_class: str | None,
+    default_to_device_class_name: bool,
+) -> tuple[dict[str, str] | None, str | None]:
+    """Translation attributes for entity with channel name."""
+    return (
+        (
+            {"channel_name": channel_name},
+            f"{key}_with_channel_name"
+            if (
+                key := translation_key
+                or (device_class if default_to_device_class_name else None)
+            )
+            else None,
+        )
+        if channel_name is not None
+        else (None, None)
+    )
```

Comment:
Could we make this more readable? Using one-liners that require several lines doesn't make sense in my opinion.